.github/workflows/reviewdog.yml

```diff
@@ -0,0 +1,41 @@
+name: VaRA Feature Library Reviewdog
+
+on:
+  pull_request:
+    branches: [ vara, vara-dev ]
+
+jobs:
+  format:
+    runs-on: ubuntu-20.04
+    continue-on-error: false
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+        with:
+          submodules: recursive
+
+      - name: Install Dependencies
+        shell: bash
+        run: |
+          sudo apt-key adv --fetch-keys https://apt.llvm.org/llvm-snapshot.gpg.key
+          sudo add-apt-repository -y 'deb http://apt.llvm.org/focal/ llvm-toolchain-focal-11 main'
+          sudo apt-get update
+          sudo apt-get -y install --no-install-recommends clang-format-11
+
+      - name: Fetch Base Branch
+        shell: bash
+        run: |
+          git fetch -n -p --depth=1 --no-recurse-submodules origin +refs/heads/${{ github.base_ref }}:refs/remotes/origin/${{ github.base_ref }}
+
+      - name: Run Clang-Format
+        shell: bash
+        run: |
+          FILES=$(git diff --name-only --diff-filter=AM origin/${{ github.base_ref }} HEAD | egrep '\.(c|h|hpp|cpp)$' || true)
+          [ -z "$FILES" ] || (clang-format-11 -i $FILES && git diff --diff-filter=M $FILES)
```

Comment:
For maintainability it would be better to have the clang-format-11 command in a variable defined above so it's easier to see where we need to update things if necessary

Reply:
How about moving `clang-format-11` and the file extension regex into an strategy, so that we are independent of types?

Reply:
I changed the job structure, so that the steps are independent of the actual tool -- which is defined as a strategy.
This allows us to use arbitrary reformatting tools for other languages (for example python) with little modification.

Reply:
very nice